components/service/glean/src/main/java/mozilla/components/service/glean/scheduler/MetricsPingScheduler.kt

```diff
@@ -0,0 +1,85 @@
+/* This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/. */
+
+package mozilla.components.service.glean.scheduler
+
+import android.content.Context
+import android.content.SharedPreferences
+import mozilla.components.support.base.log.logger.Logger
+import java.util.concurrent.TimeUnit
+
+/**
+ * MetricsPingScheduler facilitates sending and scheduling of metrics pings
+ */
+internal class MetricsPingScheduler(val applicationContext: Context) {
+    private val logger = Logger("glean/MetricsPingScheduler")
+    internal val sharedPreferences: SharedPreferences by lazy { getMetricsPrefs() }
+
+    companion object {
+        const val LAST_METRICS_PING_TIMESTAMP_KEY = "last_metrics_ping_timestamp"
+        const val STORE_NAME = "metrics"
+        const val PING_INTERVAL_HOURS = 24L
+    }
+
+    /**
+     * Check to see if a metrics ping can be sent.
+     *
+     * This will compare the current system time to the last ping time stored in [SharedPreferences]
+     * to determine whether [PING_INTERVAL_HOURS] has elapsed.
+     *
+     * @return true if the time interval has elapsed or if this is the first attempt to send a
+     *         metrics ping, otherwise false.
+     */
+    fun canSendPing(): Boolean {
+        var lastPingTime = 0L
+
+        try {
```

Comment:
Instead of using a `var`, this can be written as follows:

```
val lastPingTime = try {
 sharedPreferences.getLong(..:)
} catch (...) {
  logger...
  0L
}
```